src/app-layer-htp.c

```diff
@@ -689,6 +692,11 @@ static void HTPHandleError(HtpState *s, const uint8_t dir)
 
     size_t size = htp_list_size(s->conn->messages);
     size_t msg;
+    if(size >= HTP_MAX_MESSAGES) {
+        DEBUG_VALIDATE_BUG_ON("Too many libhtp messages");
+        // ignore further messages
```

Comment:
I think it would be good to set a 'final' event here: 'too many events' or something. Then ppl can match (or drop) on it.

Reply:
Ok